### DISCUSSION TITLE
Axes Multiple

### DISCUSSION THREAD

**Discussion by atjbat** [Live-Charts/LiveCharts2#132] [Ideas]
I' ve try to use the Axes multiple section and I' ve detected a little bug in class StrokeAndFillCartesianSeries.
The bug is : if I try to reload the chart with a new number of lines  of series then StrokeAndFillCartesianSeries go in exception because in the void : 
             public override void SoftDelete(IChartView chart) line 96
at the line :
                var primaryAxis = core.YAxes[ScalesYAt];
It try to delete the line absent because now I' ve a minor number of lines
I've put this modify:

        public override void SoftDelete(IChartView chart)
        {
            var core = ((ICartesianChartView<TDrawingContext>)chart).Core;

            if (ScalesYAt < core.XAxes.Length)
            {

                var secondaryAxis = core.XAxes[ScalesXAt];
                var primaryAxis = core.YAxes[ScalesYAt];

                var secondaryScale = new Scaler(core.DrawMarginLocation, core.DrawMarginSize, secondaryAxis);
                var primaryScale = new Scaler(core.DrawMarginLocation, core.DrawMarginSize, primaryAxis);

                var deleted = new List<ChartPoint>();
                foreach (var point in everFetched)
                {
                    if (point.Context.Chart != chart) continue;

                    SoftDeletePoint(point, primaryScale, secondaryScale);
                    deleted.Add(point);
                }

                if (Fill != null) core.Canvas.RemovePaintTask(Fill);
                if (Stroke != null) core.Canvas.RemovePaintTask(Stroke);
                if (DataLabelsPaint != null) core.Canvas.RemovePaintTask(DataLabelsPaint);

                foreach (var item in deleted) _ = everFetched.Remove(item);
            }

        }

**Comment by CodeWithMichal** [comment]
Could you please create bug report?
https://github.com/beto-rodriguez/LiveCharts2/issues

**Comment by beto-rodriguez** [comment]
Thanks for the report, I transferred this discussion to an issue so we can track and fix it.

#133

**Comment by beto-rodriguez** [comment]
@atjbat can you help me with the steps to reproduce the issue please.

How did you reload the data so the exception occurred?
